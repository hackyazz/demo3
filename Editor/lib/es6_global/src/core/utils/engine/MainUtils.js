

import * as Most from "most";
import * as RestoreJob$WonderEditor from "../../job/loop/RestoreJob.js";
import * as InitEventJob$WonderEditor from "../../job/init/InitEventJob.js";
import * as InitEditorJob$WonderEditor from "../../job/init/InitEditorJob.js";
import * as InitHotKeysJob$WonderEditor from "../../job/init/InitHotKeysJob.js";
import * as InitPickingJob$WonderEditor from "../../job/init/InitPickingJob.js";
import * as UpdateCameraJob$WonderEditor from "../../job/loop/UpdateCameraJob.js";
import * as JobEngineService$WonderEditor from "../../../service/state/engine/job/JobEngineService.js";
import * as SetOutlineDataJob$WonderEditor from "../../job/loop/SetOutlineDataJob.js";
import * as AssetEngineService$WonderEditor from "../../../service/state/engine/AssetEngineService.js";
import * as SceneEngineService$WonderEditor from "../../../service/state/engine/SceneEngineService.js";
import * as StateEditorService$WonderEditor from "../../../service/state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../../service/state/engine/state/StateEngineService.js";
import * as ParseSettingService$WonderEditor from "../../../service/record/editor/setting/ParseSettingService.js";
import * as DirectorEngineService$WonderEditor from "../../../service/state/engine/DirectorEngineService.js";
import * as InitTransformGizmosJob$WonderEditor from "../../job/init/InitTransformGizmosJob.js";
import * as ReallocateCPUMemoryJob$WonderEditor from "../../job/loop/ReallocateCPUMemoryJob.js";
import * as StateDataEngineService$WonderEditor from "../../../service/state/engine/StateDataEngineService.js";
import * as GameObjectEngineService$WonderEditor from "../../../service/state/engine/gameObject/GameObjectEngineService.js";
import * as InitCameraControllerJob$WonderEditor from "../../job/init/InitCameraControllerJob.js";
import * as SetSettingEditorService$WonderEditor from "../../../service/state/editor/setting/SetSettingEditorService.js";
import * as PrepareRenderGameViewJob$WonderEditor from "../../job/loop/PrepareRenderGameViewJob.js";
import * as RenderTransformGizmosJob$WonderEditor from "../../job/loop/RenderTransformGizmosJob.js";
import * as UpdateTransformGizmosJob$WonderEditor from "../../job/loop/UpdateTransformGizmosJob.js";
import * as PrepareRenderSceneViewJob$WonderEditor from "../../job/loop/PrepareRenderSceneViewJob.js";
import * as LoaderManagerEngineService$WonderEditor from "../../../service/state/engine/LoaderManagerEngineService.js";

function _getLoadData(param) {
  return AssetEngineService$WonderEditor.loadConfig(/* array */[
              "./config/engine/setting.json",
              "./config/engine/"
            ], StateDataEngineService$WonderEditor.getEngineStateData(/* () */0));
}

function _registerJob(engineState) {
  return JobEngineService$WonderEditor.registerNoWorkerLoopJob("restore", RestoreJob$WonderEditor.restoreJob, JobEngineService$WonderEditor.registerNoWorkerLoopJob("set_outline_data", SetOutlineDataJob$WonderEditor.setOutlineDataJob, JobEngineService$WonderEditor.registerNoWorkerLoopJob("prepare_render_game_view", PrepareRenderGameViewJob$WonderEditor.prepareRenderGameViewJob, JobEngineService$WonderEditor.registerNoWorkerLoopJob("prepare_render_scene_view", PrepareRenderSceneViewJob$WonderEditor.prepareRenderSceneViewJob, JobEngineService$WonderEditor.registerNoWorkerLoopJob("render_transform_gizmos", RenderTransformGizmosJob$WonderEditor.renderJob, JobEngineService$WonderEditor.registerNoWorkerLoopJob("update_transform_gizmos", UpdateTransformGizmosJob$WonderEditor.updateTransformJob, JobEngineService$WonderEditor.registerNoWorkerLoopJob("update_camera", UpdateCameraJob$WonderEditor.updateJob, JobEngineService$WonderEditor.registerNoWorkerLoopJob("reallocate_cpu_memory", ReallocateCPUMemoryJob$WonderEditor.reallocateJob, JobEngineService$WonderEditor.registerNoWorkerInitJob("init_camera_controller", InitCameraControllerJob$WonderEditor.initJob, JobEngineService$WonderEditor.registerNoWorkerInitJob("init_transform_gizmos", InitTransformGizmosJob$WonderEditor.initJob, JobEngineService$WonderEditor.registerNoWorkerInitJob("init_picking", InitPickingJob$WonderEditor.initJob, JobEngineService$WonderEditor.registerNoWorkerInitJob("init_transform_gizmos", InitTransformGizmosJob$WonderEditor.initJob, JobEngineService$WonderEditor.registerNoWorkerInitJob("init_hotkeys", InitHotKeysJob$WonderEditor.initHotKeysForEditorJob, JobEngineService$WonderEditor.registerNoWorkerInitJob("init_event_for_editor", InitEventJob$WonderEditor.initEventForEditorJob, JobEngineService$WonderEditor.registerNoWorkerInitJob("init_editor", InitEditorJob$WonderEditor.initEditorJob, engineState)))))))))))))));
}

function _handleEngineState(engineState) {
  var engineState$1 = _registerJob(engineState);
  var scene = SceneEngineService$WonderEditor.getSceneGameObject(engineState$1);
  return StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.init(GameObjectEngineService$WonderEditor.setGameObjectName(SceneEngineService$WonderEditor.getDefaultName(/* () */0), scene, engineState$1)));
}

function init(param) {
  return Most.drain(Most.map(_handleEngineState, Most.flatMap((function (engineState) {
                        return Most.fromPromise(fetch("./config/editor/setting.json").then((function (response) {
                                          return response.json().then((function (json) {
                                                        var __x = ParseSettingService$WonderEditor.convertToRecord(json);
                                                        StateEditorService$WonderEditor.setState(SetSettingEditorService$WonderEditor.setSetting(__x, StateEditorService$WonderEditor.getState(/* () */0)));
                                                        return Promise.resolve(engineState);
                                                      }));
                                        })));
                      }), Most.flatMap((function (engineState) {
                            return Most.fromPromise(LoaderManagerEngineService$WonderEditor.loadIMGUIAsset("./public/font/empty.fnt", "./public/font/empty.png", /* array */[
                                            /* tuple */[
                                              "./public/img/camera.png",
                                              "camera"
                                            ],
                                            /* tuple */[
                                              "./public/img/sun.png",
                                              "directionLight"
                                            ],
                                            /* tuple */[
                                              "./public/img/point.png",
                                              "pointLight"
                                            ]
                                          ], (function (param, param$1) {
                                              return /* () */0;
                                            }), engineState));
                          }), _getLoadData(/* () */0)))));
}

function start(param) {
  return init(/* () */0);
}

export {
  _getLoadData ,
  _registerJob ,
  _handleEngineState ,
  init ,
  start ,
  
}
/* most Not a pure module */
