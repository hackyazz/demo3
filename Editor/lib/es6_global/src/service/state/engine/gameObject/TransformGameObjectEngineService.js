

import * as TransformEngineService$WonderEditor from "../TransformEngineService.js";
import * as GameObjectComponentEngineService$WonderEditor from "./GameObjectComponentEngineService.js";

function getPosition(gameObject, engineState) {
  return TransformEngineService$WonderEditor.getPosition(GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(gameObject, engineState), engineState);
}

function getLocalPosition(gameObject, engineState) {
  return TransformEngineService$WonderEditor.getLocalPosition(GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(gameObject, engineState), engineState);
}

function setLocalPosition(gameObject, pos, engineState) {
  return TransformEngineService$WonderEditor.setLocalPosition(pos, GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(gameObject, engineState), engineState);
}

function getLocalScale(gameObject, engineState) {
  return TransformEngineService$WonderEditor.getLocalScale(GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(gameObject, engineState), engineState);
}

function setLocalScale(gameObject, scale, engineState) {
  return TransformEngineService$WonderEditor.setLocalScale(scale, GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(gameObject, engineState), engineState);
}

function getLocalToWorldMatrixTypeArray(gameObject, engineState) {
  return TransformEngineService$WonderEditor.getLocalToWorldMatrixTypeArray(GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(gameObject, engineState), engineState);
}

export {
  getPosition ,
  getLocalPosition ,
  setLocalPosition ,
  getLocalScale ,
  setLocalScale ,
  getLocalToWorldMatrixTypeArray ,
  
}
/* TransformEngineService-WonderEditor Not a pure module */
