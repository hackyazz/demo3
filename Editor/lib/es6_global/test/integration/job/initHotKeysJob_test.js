

import * as Curry from "../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as TestTool$WonderEditor from "../../tool/TestTool.js";
import * as EventTool$WonderEditor from "./tool/EventTool.js";
import * as MainUtils$WonderEditor from "../../../src/core/utils/engine/MainUtils.js";
import * as FloatService$WonderEditor from "../../../src/service/atom/FloatService.js";
import * as ReactTestTool$WonderEditor from "../../tool/ReactTestTool.js";
import * as GameObjectTool$WonderEditor from "../../tool/GameObjectTool.js";
import * as Vector3Service$WonderEditor from "../../../src/service/primitive/Vector3Service.js";
import * as KeyboardEventTool$WonderEditor from "./tool/KeyboardEventTool.js";
import * as StateLogicService$WonderEditor from "../../../src/service/stateTuple/logic/StateLogicService.js";
import * as BuildComponentTool$WonderEditor from "../../tool/BuildComponentTool.js";
import * as StateEngineService$WonderEditor from "../../../src/service/state/engine/state/StateEngineService.js";
import * as MainEditorSceneTool$WonderEditor from "../../tool/MainEditorSceneTool.js";
import * as SceneViewEditorService$WonderEditor from "../../../src/service/state/editor/view/sceneView/SceneViewEditorService.js";
import * as TransformEngineService$WonderEditor from "../../../src/service/state/engine/TransformEngineService.js";
import * as MainEditorLeftHeaderTool$WonderEditor from "../../unit/composable_component/mainEditor/composable_component/leftHeader/tool/MainEditorLeftHeaderTool.js";
import * as ArcballCameraEngineService$WonderEditor from "../../../src/service/state/engine/ArcballCameraEngineService.js";
import * as NoWorkerJobConfigToolEngine$WonderEditor from "../../tool/engine/NoWorkerJobConfigToolEngine.js";
import * as GameObjectComponentEngineService$WonderEditor from "../../../src/service/state/engine/gameObject/GameObjectComponentEngineService.js";
import * as TransformGameObjectEngineService$WonderEditor from "../../../src/service/state/engine/gameObject/TransformGameObjectEngineService.js";

describe("init hotKeys job", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        var _prepareKeyboardEvent = function (sandbox, _) {
          return MainEditorSceneTool$WonderEditor.initStateWithJob(sandbox, NoWorkerJobConfigToolEngine$WonderEditor.buildNoWorkerJobConfig(undefined, "\n                [\n                  {\n                    \"name\": \"default\",\n                    \"jobs\": [\n                      {\n                        \"name\": \"init_hotkeys\"\n                      }\n                    ]\n                  }\n                ]\n            ", "\n                     [\n                         {\n                             \"name\": \"default\",\n                             \"jobs\": [\n                                 {\n                                     \"name\": \"dispose\"\n                                 }\n                             ]\n                         }\n                     ]\n                 ", "\n                [\n\n                    {\n                       \"name\": \"init_hotkeys\"\n                    }\n                ]\n            ", undefined, /* () */0), undefined, undefined, false, undefined, /* () */0);
        };
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                return /* () */0;
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("bind document hotKeys event", (function () {
                var _execKeyboardEvent = function (keyboardDomEventName, keyCode, $staropt$star, $staropt$star$1, $staropt$star$2, _) {
                  var ctrlKey = $staropt$star !== undefined ? $staropt$star : false;
                  var altKey = $staropt$star$1 !== undefined ? $staropt$star$1 : false;
                  var shiftKey = $staropt$star$2 !== undefined ? $staropt$star$2 : false;
                  return EventTool$WonderEditor.triggerDomEvent(keyboardDomEventName, EventTool$WonderEditor.getDocument(/* () */0), KeyboardEventTool$WonderEditor.buildKeyboardEvent(ctrlKey, altKey, shiftKey, undefined, keyCode, /* () */0));
                };
                var triggerRedoHotKeyEvent = function () {
                  _execKeyboardEvent("keydown", 89, true, undefined, undefined, /* () */0);
                  return /* () */0;
                };
                var triggerUndoHotKeyEvent = function () {
                  _execKeyboardEvent("keydown", 90, true, undefined, undefined, /* () */0);
                  return /* () */0;
                };
                var triggerCloneHotKeyEvent = function () {
                  _execKeyboardEvent("keydown", 68, true, undefined, undefined, /* () */0);
                  return /* () */0;
                };
                var triggerDeleteHotKeyEvent = function () {
                  _execKeyboardEvent("keydown", 46, undefined, undefined, undefined, /* () */0);
                  return /* () */0;
                };
                var triggerFocusHotKeyEvent = function () {
                  _execKeyboardEvent("keydown", 70, undefined, undefined, undefined, /* () */0);
                  return /* () */0;
                };
                beforeEach((function () {
                        _prepareKeyboardEvent(sandbox, /* () */0);
                        MainEditorSceneTool$WonderEditor.createDefaultSceneAndNotInit(sandbox);
                        Curry._1(EventTool$WonderEditor.buildFakeDocumentSetToWindow, /* () */0);
                        return StateLogicService$WonderEditor.getAndSetEngineState(MainUtils$WonderEditor._handleEngineState);
                      }));
                afterEach((function () {
                        return EventTool$WonderEditor.restoreHotKeys(/* () */0);
                      }));
                describe("test bind undo hot-key", (function () {
                        return Wonder_jest.test("key down ctrl+z, should execute undo operate", (function () {
                                      MainEditorLeftHeaderTool$WonderEditor.addEmptyGameObject(undefined, undefined, /* () */0);
                                      triggerUndoHotKeyEvent(/* () */0);
                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildSceneTree(TestTool$WonderEditor.buildEmptyAppState(/* () */0)));
                                    }));
                      }));
                describe("test bind clone hot-key", (function () {
                        return Wonder_jest.test("key down ctrl+d, should execute clone operate", (function () {
                                      triggerCloneHotKeyEvent(/* () */0);
                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildSceneTree(TestTool$WonderEditor.buildEmptyAppState(/* () */0)));
                                    }));
                      }));
                describe("test bind redo hot-key", (function () {
                        return Wonder_jest.test("key down ctrl+y, should execute redo operate", (function () {
                                      MainEditorLeftHeaderTool$WonderEditor.addEmptyGameObject(undefined, undefined, /* () */0);
                                      MainEditorLeftHeaderTool$WonderEditor.addEmptyGameObject(undefined, undefined, /* () */0);
                                      triggerUndoHotKeyEvent(/* () */0);
                                      triggerUndoHotKeyEvent(/* () */0);
                                      triggerRedoHotKeyEvent(/* () */0);
                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildSceneTree(TestTool$WonderEditor.buildEmptyAppState(/* () */0)));
                                    }));
                      }));
                describe("test bind delete hot-key", (function () {
                        return Wonder_jest.test("key down delete,if has currentSceneTreeNode, should delete it", (function () {
                                      MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode(/* () */0);
                                      triggerDeleteHotKeyEvent(/* () */0);
                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildSceneTree(TestTool$WonderEditor.buildEmptyAppState(/* () */0)));
                                    }));
                      }));
                describe("test bind focus hot-key", (function () {
                        var _getDistance = function (param) {
                          var engineState = param[1];
                          var __x = ArcballCameraEngineService$WonderEditor.unsafeGetArcballCameraControllerDistance(GameObjectComponentEngineService$WonderEditor.unsafeGetArcballCameraControllerComponent(SceneViewEditorService$WonderEditor.unsafeGetEditCamera(param[0]), engineState), engineState);
                          return FloatService$WonderEditor.truncateFloatValue(__x, 3);
                        };
                        var _getTarget = function (param) {
                          var engineState = param[1];
                          return Vector3Service$WonderEditor.truncate(3, ArcballCameraEngineService$WonderEditor.unsafeGetArcballCameraControllerTarget(GameObjectComponentEngineService$WonderEditor.unsafeGetArcballCameraControllerComponent(SceneViewEditorService$WonderEditor.unsafeGetEditCamera(param[0]), engineState), engineState));
                        };
                        describe("\n          calc currentSceneTreeNode's all children and its self->aabb;\n          use aabb's center as arcball camera controller target;\n          use aabb's radius * factor as arcball camera controller distance;\n          ", (function () {
                                Wonder_jest.test("test the currentSceneTreeNode is scene gameObject", (function () {
                                        GameObjectTool$WonderEditor.setCurrentSceneTreeNode(MainEditorSceneTool$WonderEditor.unsafeGetScene(/* () */0));
                                        var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                        var engineState$1 = TransformEngineService$WonderEditor.setScale(GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(MainEditorSceneTool$WonderEditor.unsafeGetScene(/* () */0), engineState), /* tuple */[
                                              3,
                                              1,
                                              1
                                            ], engineState);
                                        var firstChild = MainEditorSceneTool$WonderEditor.getFirstCube(engineState$1);
                                        var secondChild = MainEditorSceneTool$WonderEditor.getSecondCube(engineState$1);
                                        var engineState$2 = TransformGameObjectEngineService$WonderEditor.setLocalPosition(secondChild, /* tuple */[
                                              -3,
                                              0,
                                              0
                                            ], TransformGameObjectEngineService$WonderEditor.setLocalPosition(firstChild, /* tuple */[
                                                  2,
                                                  0,
                                                  0
                                                ], engineState$1));
                                        StateEngineService$WonderEditor.setState(engineState$2);
                                        triggerFocusHotKeyEvent(/* () */0);
                                        return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](/* tuple */[
                                                        StateLogicService$WonderEditor.getStateToGetData(_getDistance),
                                                        StateLogicService$WonderEditor.getStateToGetData(_getTarget)
                                                      ]), /* tuple */[
                                                    22.569,
                                                    /* tuple */[
                                                      -1.5,
                                                      0,
                                                      0
                                                    ]
                                                  ]);
                                      }));
                                return Wonder_jest.test("test the currentSceneTreeNode is scene children", (function () {
                                              MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode(/* () */0);
                                              var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                              var firstChild = MainEditorSceneTool$WonderEditor.getFirstCube(engineState);
                                              var secondChild = MainEditorSceneTool$WonderEditor.getSecondCube(engineState);
                                              var engineState$1 = TransformEngineService$WonderEditor.setScale(GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(firstChild, engineState), /* tuple */[
                                                    3,
                                                    1,
                                                    1
                                                  ], engineState);
                                              var engineState$2 = TransformGameObjectEngineService$WonderEditor.setLocalPosition(secondChild, /* tuple */[
                                                    -3,
                                                    0,
                                                    0
                                                  ], TransformGameObjectEngineService$WonderEditor.setLocalPosition(firstChild, /* tuple */[
                                                        2,
                                                        0,
                                                        0
                                                      ], engineState$1));
                                              StateEngineService$WonderEditor.setState(engineState$2);
                                              triggerFocusHotKeyEvent(/* () */0);
                                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](/* tuple */[
                                                              StateLogicService$WonderEditor.getStateToGetData(_getDistance),
                                                              StateLogicService$WonderEditor.getStateToGetData(_getTarget)
                                                            ]), /* tuple */[
                                                          4.146,
                                                          /* tuple */[
                                                            2,
                                                            0,
                                                            0
                                                          ]
                                                        ]);
                                            }));
                              }));
                        describe("fix bug", (function () {
                                describe("if currentSceneTreeNode and its all children has no geometry component", (function () {
                                        var _prepareAndExec = function () {
                                          GameObjectTool$WonderEditor.setCurrentSceneTreeNode(StateLogicService$WonderEditor.getEngineStateToGetData(MainEditorSceneTool$WonderEditor.getCameraInDefaultScene));
                                          var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                          var camera = MainEditorSceneTool$WonderEditor.getCameraInDefaultScene(engineState);
                                          var pos = /* tuple */[
                                            2,
                                            0,
                                            0
                                          ];
                                          var engineState$1 = TransformGameObjectEngineService$WonderEditor.setLocalPosition(camera, pos, engineState);
                                          StateEngineService$WonderEditor.setState(engineState$1);
                                          triggerFocusHotKeyEvent(/* () */0);
                                          return pos;
                                        };
                                        Wonder_jest.test("use currentSceneTreeNode->position as target", (function () {
                                                var pos = _prepareAndExec(/* () */0);
                                                return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](StateLogicService$WonderEditor.getStateToGetData(_getTarget)), pos);
                                              }));
                                        return Wonder_jest.test("use fixed value as distance", (function () {
                                                      _prepareAndExec(/* () */0);
                                                      return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](StateLogicService$WonderEditor.getStateToGetData(_getDistance)), 3);
                                                    }));
                                      }));
                                return /* () */0;
                              }));
                        return /* () */0;
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
