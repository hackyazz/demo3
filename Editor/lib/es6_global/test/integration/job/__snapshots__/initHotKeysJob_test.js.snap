// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init hotKeys job bind document hotKeys event test bind clone hot-key key down ctrl+d, should execute clone operate 1`] = `
<article
  className="wonder-sceneTree-component"
  id="wonder-sceneTree-component"
>
  <article
    className="wonder-tree"
  >
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
          onMouseDown={[Function]}
        >
          <img
            src="./public/img/down.png"
          />
        </div>
        <div
          className="draggable-container  no-drag"
          draggable={false}
          id="sceneTreeNode-0"
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          Scene
        </div>
      </li>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-6"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Camera
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-3"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Cube
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-4"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Cube
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-5"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Direction Light
          </div>
        </li>
      </ul>
    </ul>
  </article>
</article>
`;

exports[`init hotKeys job bind document hotKeys event test bind delete hot-key key down delete,if has currentSceneTreeNode, should delete it 1`] = `
<article
  className="wonder-sceneTree-component"
  id="wonder-sceneTree-component"
>
  <article
    className="wonder-tree"
  >
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
          onMouseDown={[Function]}
        >
          <img
            src="./public/img/down.png"
          />
        </div>
        <div
          className="draggable-container  no-drag"
          draggable={false}
          id="sceneTreeNode-0"
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          Scene
        </div>
      </li>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-6"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Camera
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-4"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Cube
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-5"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Direction Light
          </div>
        </li>
      </ul>
    </ul>
  </article>
</article>
`;

exports[`init hotKeys job bind document hotKeys event test bind redo hot-key key down ctrl+y, should execute redo operate 1`] = `
<article
  className="wonder-sceneTree-component"
  id="wonder-sceneTree-component"
>
  <article
    className="wonder-tree"
  >
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
          onMouseDown={[Function]}
        >
          <img
            src="./public/img/down.png"
          />
        </div>
        <div
          className="draggable-container  no-drag"
          draggable={false}
          id="sceneTreeNode-0"
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          Scene
        </div>
      </li>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-6"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Camera
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-3"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Cube
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-4"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Cube
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-5"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Direction Light
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-7"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            GameObject
          </div>
        </li>
      </ul>
    </ul>
  </article>
</article>
`;

exports[`init hotKeys job bind document hotKeys event test bind undo hot-key key down ctrl+z, should execute undo operate 1`] = `
<article
  className="wonder-sceneTree-component"
  id="wonder-sceneTree-component"
>
  <article
    className="wonder-tree"
  >
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
          onMouseDown={[Function]}
        >
          <img
            src="./public/img/down.png"
          />
        </div>
        <div
          className="draggable-container  no-drag"
          draggable={false}
          id="sceneTreeNode-0"
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          Scene
        </div>
      </li>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-6"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Camera
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-3"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Cube
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-4"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Cube
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-5"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Direction Light
          </div>
        </li>
      </ul>
    </ul>
  </article>
</article>
`;
